plugin/evm/vm.go

```diff
@@ -1055,6 +1055,9 @@ func (vm *VM) SetState(_ context.Context, state snow.State) error {
 		if err := vm.StateSyncClient.Error(); err != nil {
 			return err
 		}
+		// Ensure snapshots are initialized before bootstrapping (i.e., if state sync is skipped).
+		// Note calling this function has no effect if snapshots are already initialized.
+		vm.blockChain.InitializeSnapshots()
```

Comment:
To ask a dumb question....we always enter bootstrapping at least once, right?

Reply:
I believe so, but @StephenButtolph knows this kind of invariant better